pkg/reconcilermanager/controllers/otel_controller.go

```diff
@@ -140,8 +140,12 @@ func (r *OtelReconciler) configureGooglecloudConfigMap(ctx context.Context) ([]b
 			metadata.SystemLabel: "true",
 			metadata.ArchLabel:   "csmr",
 		}
+		configYAML, yamlErr := metrics.CollectorConfigGooglecloudYAML()
+		if yamlErr != nil {
+			return status.APIServerErrorf(yamlErr, "failed to marshal otel-collector-googlecloud config to YAML")
```

Comment:
Is this an API server error?

Reply:
Missed the message. Changed to regular error for marshaling.